### PR TITLE
[ENH] Add Full ETS forecaster

### PR DESCRIPTION
**Pull request by alexbanwell1**
#### Reference Issues/PRs
See also #2244

#### What does this implement/fix? Explain your changes.
Adds full functionality to ETS model. Adds additional tests for this. Removes default argument form BaseForecaster.forecast method, as it is required to call BaseForecaster.fit(). Add code to verify ETS Forecaster against stats model.

#### Does your contribution introduce a new dependency? If yes, which one?
It uses the statsforecast package to test against, it is optional

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/aeon-toolkit/aeon/blob/main/.all-contributorsrc). Alternatively, you can use the [@all-contributors](https://allcontributors.org/docs/en/bot/usage) bot to do this for you.
- [x] The PR title starts with either [ENH], [MNT], [DOC], [BUG], [REF], [DEP] or [GOV] indicating whether the PR topic is related to enhancement, maintenance, documentation, bugs, refactoring, deprecation or governance.

##### For new estimators and functions
- [ ] I've added the estimator to the online [API documentation](https://www.aeon-toolkit.org/en/latest/api_reference.html).

### CODE REVIEW

**Comment by alexbanwell1** [comment]
@TonyBagnall

File: aeon/forecasting/_ets.py [hunk 1]
```diff
@@ -1,35 +1,129 @@
...
 import numpy as np
 
 from aeon.forecasting.base import BaseForecaster
 
+NONE = 0
+ADDITIVE = 1
+MULTIPLICATIVE = 2
+
+
+class ModelType:
```
Comment (TonyBagnall) [inline aeon/forecasting/_ets.py:25]: I think this should be protected and not in all, (_ModelType) as it is specific to ETS

File: aeon/forecasting/_ets.py [hunk 2]
```diff
@@ -1,35 +1,129 @@
...
+    """
+    Class describing the error, trend and seasonality model of an ETS forecaster.
```
Comment (TonyBagnall) [inline aeon/forecasting/_ets.py:27]: given we are going to pull it all out of the object for numba, I'm not sure there is much point encapsulating the configuration like this. We can see how it pans out

**Review by TonyBagnall — APPROVED** [review]
looks good for the branch, we can discuss a design issues

**Pull request merged**